src/provider/provider_coarse.c

```diff
@@ -1643,9 +1642,7 @@ static umf_result_t coarse_memory_provider_allocation_merge(void *provider,
                                                             void *lowPtr,
                                                             void *highPtr,
                                                             size_t totalSize) {
-    if (provider == NULL || lowPtr == NULL || highPtr == NULL ||
-        ((uintptr_t)highPtr <= (uintptr_t)lowPtr) ||
-        ((uintptr_t)highPtr - (uintptr_t)lowPtr >= totalSize)) {
```

Comment:
just asking, to confirm - the check is slightly different in `src/memory_provider.c`: `(uintptr_t)highPtr - (uintptr_t)lowPtr > totalSize`

Reply:
The check fixed in `src/memory_provider.c`
Done.

Reply:
@lukaszstolarczuk ping

Reply:
@lukaszstolarczuk ping